### DISCUSSION TITLE
Noise at the end of produced wave file

### DISCUSSION THREAD

**Discussion by iarspider** [snakers4/silero-models#93] [Q&A] [answered]
## 🐛 Bug

Noise at the end of produced wave file

## To Reproduce

1. Install torch, numpy
2. Run the example for Habrahabr article:
```py
import os
import torch

device = torch.device('cpu')
torch.set_num_threads(4)
local_file = 'model.pt'

if not os.path.isfile(local_file):
    torch.hub.download_url_to_file('https://models.silero.ai/models/tts/ru/v2_kseniya.pt',
                                   local_file)  

model = torch.package.PackageImporter(local_file).load_pickle("tts_models", "model")
model.to(device)

example_batch = ['В недрах тундры выдры в г+етрах т+ырят в вёдра ядра кедров.',
                 'Котики - это жидкость!',
                 'М+ама М+илу м+ыла с м+ылом.']
sample_rate = 16000

audio_paths = model.save_wav(texts=example_batch,
                             sample_rate=sample_rate)
```
3. The second and third audio files are 12 seconds long (instead of ~1 sec), and are "padded" with noise
![image](https://user-images.githubusercontent.com/636602/122553195-9fefbe80-d037-11eb-8d73-4b8bf6f9db07.png)

## Expected behavior

No padding with noise

## Environment

Please copy and paste the output from this 
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

```
Collecting environment information...
PyTorch version: 1.9.0+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.8 (64-bit runtime)
Python platform: Windows-10-10.0.18362-SP0
Is CUDA available: False
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080
Nvidia driver version: 466.77
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.3
[pip3] torch==1.9.0
[conda] Could not collect
```

## Additional context

Thanks a lot for creating this!

**Comment by snakers4** [comment, marked as answer]
This is a known tacotron bug
Happens due to batching

Batch size 1 fixes this (i.e. just feed 1 phrase)

We basically have 2 options how to fix this more properly

- Downstream with a VAD (please use `silero-vad` or WebRTC)
- By looking at the alignment map of tacotron (we did not implement this yet)

**Comment by snakers4** [comment]
in future though we will stop using tacotron, so the maybe just using batch_size = 1 or just applying a VAD is good temporary solution (especially when you run on CPU)
